app/models/loan.rb

```diff
@@ -235,4 +235,9 @@ def project_events(order_by="Completed IS NULL OR Completed = '0000-00-00', Comp
   def logs
     project_logs
   end
+
+  def calendar_events
+    [CalendarEvent.new_loan_start(self), CalendarEvent.new_loan_end(self)].compact
```

Comment:
Hmm, what you've done here is spread the responsibility for creating calendar events over two classes -- this one needs to know that a loan start and loan end event are required, and the cal event class knows what the params should be based on object type. I think it should be one or the other. So I would just do like `CalendarEvent.build_for(self)`, then CalendarEvent figures it all out.